packages/network-controller/src/NetworkController.ts

```diff
@@ -333,6 +337,13 @@ type CustomNetworkClientId = string;
  */
 type NetworkClientId = BuiltInNetworkClientId | CustomNetworkClientId;
 
+/**
+ * Data related to a particular network.
+ */
+export type NetworksMetadata = {
+  [networkClientId: NetworkClientId]: NetworkDetails;
```

Comment:
What are our thoughts on renaming `NetworkDetails` to `NetworkMetadata` to match this type? Alternatively, we could rename `NetworksMetadata` to `NetworksDetails`. It seems like it would be confusing if they had different names otherwise.